runner/internal/shim/task.go

```diff
@@ -45,19 +46,19 @@ type Task struct {
 
 // Lock is used for exclusive operations, e.g, stopping a container,
 // removing task data, etc.
-func (t *Task) Lock() {
+func (t *Task) Lock(ctx context.Context) {
 	if !t.mu.TryLock() {
-		log.Fatalf("task %s already locked!", t.ID)
+		log.Fatal(ctx, "already locked!", "task", t.ID)
```

Comment:
log.Fatal calls exit(1) and defers do not run, i.e. basically panics. Is it intended here?

Reply:
> log.Fatal calls exit(1) and defers do not run

Didn't know that, but in this specific case it's fine I think, this should never happen, fatal here indicates a serious  programmatic error (which otherwise would lead to a deadlock, as mutexes are not reentrant) and should abort execution, even if not gracefully